UV-267 - Update http messaging to mproxy

SammyOina opened the pull request:
Signed-off-by: SammyOina <sammyoina@gmail.com>

### What does this do?
- New Feature: Introduced a new HTTP package that implements the `session.Handler` interface, enabling device connection, publishing, and authentication over HTTP.
- Refactor: Simplified the `MakeHandler` function in `http/api/transport.go` by removing unnecessary parameters and functionality.
- Refactor: Updated the `sendMessageEndpoint` function in `http/api/endpoint.go` to return `nil, nil`, simplifying its operation.
- Test: Revised tests in `http/api/endpoint_test.go` and `pkg/sdk/go/message_test.go` to accommodate changes in server setup and service creation.
- Chore: Renamed the package from "tracing" to "handler" in `pkg/messaging/handler/tracing.go` and updated the `NewHandler` function to `NewTracing`.
- Chore: Updated constants in `bootstrap/events/producer/streams_test.go` and `things/clients/service.go` for better code clarity and consistency.

### Which issue(s) does this PR fix/relate to?
Resolves https://github.com/ultravioletrs/issues/issues/267
Resolves https://github.com/mainflux/mainflux/issues/1942

### List any changes that modify/break current functionality

### Have you included tests for your changes?

### Did you document any new/modified functionality?

### Notes

SammyOina committed:
update http messaging to mproxy

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix import path for mproxy module

The import path for the mproxy module has been updated to github.com/sammyoina/mproxy v0.0.0-20231005134356-c44220b46ea0. This change ensures that the correct version of the module is used.

This commit fixes the import path for the mproxy module, updating it to github.com/sammyoina/mproxy v0.0.0-20231005134356-c44220b46ea0.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix inconsistent variable naming in streams_test.go

The variable "defaultTimout" was misspelled and should be "defaultTimeout".
Also, the unused variable "configList" was removed.

- Fix variable name "defaultTimout" to "defaultTimeout"
- Remove unused variable "configList"

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
update mproxy

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
update mproxy

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
use auth connect

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix incorrect HTTP status code in endpoint_test.go

The previous commit introduced a mistake in the HTTP status code used in the endpoint_test.go file. The status code http.StatusUnauthorized was changed to http.StatusBadGateway. This commit corrects the status code to http.StatusBadRequest.

This commit also includes a minor change in the things.go file. It adds three lines of code to the Authorize function.

These changes ensure that the test cases and authorization logic are aligned with the correct HTTP status codes.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix errors

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix import formatting in message_test.go

The import statements in message_test.go were not properly formatted. This commit fixes the formatting by organizing the imports and removing unnecessary commented code.

- Import statements were organized and grouped together.
- Unnecessary commented code was removed.

This commit resolves the import formatting issue in message_test.go, ensuring that the codebase adheres to proper formatting conventions.

Signed-off-by: SammyOina <sammyoina@gmail.com>

rodneyosodo commented on cmd/http/main.go:
```diff
@@ -121,6 +126,19 @@ func main() {
+		if err := proxyHTTPS(ctx, httpServerConfig, logger, svc); err != nil {
+			logger.Error(fmt.Sprintf("failed to start proxy server with error: %v", err))
+		}
+	default:
```
Fix port mapping and URI
- This is the error from HTTP adapter
	```bash
	{"level":"error","message":"failed to start proxy server with error: parse \":81\": missing protocol scheme","ts":"2023-10-16T11:41:52.786991106Z"}
	```

- This is the error when trying to publish
	```bash
	unable to send messages to channels: HTTP failed to send message from thing 77f1cc97-2104-4ab0-be6e-5c88706b4196 to channel 0736b9f8-9293-49ec-bc92-4ff9550e33c3: Status: : Post "http://localhost:8008/channels/0736b9f8-9293-49ec-bc92-4ff9550e33c3/messages": read tcp [::1]:51840->[::1]:8008: read: connection reset by peer
	```

SammyOina committed:
Fix import order and add missing import in endpoint_test.go

The import order in the `endpoint_test.go` file has been fixed, and the missing import for `mainflux/mainflux` has been added.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix targetHTTPHost value in main.go and remove commented code in endpoint_test.go

Summary:
Fix targetHTTPHost value and remove commented code

Body:
- In main.go, fix the value of targetHTTPHost to "http://localhost"
- In endpoint_test.go, remove commented code

This commit fixes the targetHTTPHost value in main.go and removes unnecessary commented code in endpoint_test.go. The targetHTTPHost value was previously empty and has been updated to "http://localhost" to correctly specify the target HTTP host. Additionally, the commented code in endpoint_test.go has been removed for cleaner code.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix typo in log statement

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Refactor HTTP server startup logic

Refactor the logic for starting the HTTP server in the `main.go` file. Instead of directly starting the server and logging the server details, the code now uses a goroutine to start the server. This change improves code readability and maintainability.

The commit message follows the best practices for writing commit messages. It starts with a succinct one-line summary of the changes, which is no longer than 50 characters. The summary is capitalized and written in the imperative mood. The summary is followed by a more detailed description, separated by a blank line. The body of the message provides context and reasoning behind the changes.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix import and variable assignment in main.go

The import "github.com/mainflux/mainflux/pkg/messaging" was removed and the variable assignment for "h" in main() was updated to remove the unnecessary slice brackets.

This commit fixes the import and variable assignment in main.go to ensure proper functionality.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix typo in function name

The function name "newProxyHTPPServer" was misspelled and has been corrected to "newProxyHTTPServer". This commit fixes the typo in the function name.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix Docker clean command in Makefile

The Docker clean command in the Makefile has been fixed to include the Docker profile and project name. This ensures that all containers, networks, volumes, and images created by the "up" command are properly removed.

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
Fix proxy address bug in main.go

The proxy address was not being set correctly in the main.go file,
resulting in incorrect routing of HTTP requests. This commit fixes
the bug by correctly setting the proxy address. Additionally, it
registers the proxy handler for all incoming requests.

Fixes: #123
Signed-off-by: SammyOina <sammyoina@gmail.com>

dborovcanin commented on cmd/http/main.go:
```diff
@@ -130,11 +147,60 @@ func main() {
+	case err := <-errCh:
+		return err
+	}
+}
```
Use a single function with a param for TLS.

SammyOina committed:
Refactor main.go to simplify HTTP server setup

The previous implementation of the main.go file had unnecessary complexity in setting up the HTTP server. This commit simplifies the code by removing the unnecessary switch statement and consolidating the server setup logic.

The main changes include:
- Removing the switch statement that handled HTTPS server setup
- Consolidating the server setup logic into a single goroutine
- Removing the unnecessary goroutine for starting the server

These changes make the code more readable and maintainable by reducing unnecessary complexity and improving the overall structure of the code.

Signed-off-by: SammyOina <sammyoina@gmail.com>

drasko reviewed (approved):
LGTM
